docs/ui/Toc.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Link, List } from '@marigold/components';
+
+export interface TocProps {
+  data: string;
+}
+
+type Item = {
+  anchor: string;
+  title: string;
+  id: string;
+  level: string;
+};
+
+export const Toc = ({ data }: TocProps) => {
+  const elements = JSON.parse(data) as Item[];
+
+  const [, setIsMounted] = useState(false);
+
+  const ref = useRef<Element>();
+
+  const itemId = elements.map(item => item.id);
+  const activeItem = useActiveItem(itemId);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      ref.current = document.querySelector('#toc') || undefined;
+      setIsMounted(true);
+    }
+  }, []);
+
+  if (!ref.current || elements.length === 0) {
+    return null;
+  }
+
+  const TocPortal = () => (
+    <div className="fixed">
+      On This Page
```

Comment:
wrap in a headline? :)

Reply:
I now wrapped it in a `Text` so it has the same height and look of the Side Navigation Headline ? Which headline would you use

Reply:
IDK not too large. Text migt be also fine